build/detect-upstream.sh

```diff
@@ -0,0 +1,93 @@
+#! /bin/bash
+
+set -e
+
+make_flag=''
+imports_flag=''
+
+while getopts 'm:i:' flag; do
+  case "${flag}" in
+    m) make_flag="${OPTARG}" ;;
+    i) imports_flag="${OPTARG}" ;;
+  esac
+done
+
+importfailure=0
+makefailure=0
+
+ignore=(
+  "^open-cluster-management\.io\/addon-framework"
+  "^open-cluster-management\.io\/api"
+  "^open-cluster-management\.io\/multicloud-operators-channel"
+)
+
+parse_imports(){
+  echo "Scanning dependencies..."
+
+  gh_ocm='^github\.com\/open-cluster-management'
+  ocm='^open-cluster-management\.io'
+  replaced='.*\=>.github\.com\/stolostron'
```

Comment:
The dot before `github` should likely be any white space character rather than any character.